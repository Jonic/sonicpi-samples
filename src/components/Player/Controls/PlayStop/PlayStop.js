import React, { Component } from 'react'
import PropTypes from 'prop-types'

class PlayStop extends Component {
  determineLabel = () => {
    return this.props.isPlaying ? 'Stop' : 'Play'
  }

  handleClick = () => {
    this.props.playStopClickHandler()
  }

  render() {
    return (
      <div className="c-PlayStop">
        <button
          className="c-PlayStop__button"
          onClick={this.handleClick}
        >
          {this.determineLabel()}
        </button>
      </div>
    )
  }
}

PlayStop.propTypes = {
  isPlaying:            PropTypes.bool.isRequired,
  playStopClickHandler: PropTypes.func.isRequired,
}

export default PlayStop
