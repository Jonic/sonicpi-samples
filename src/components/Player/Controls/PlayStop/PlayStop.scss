@import '../../../../styles/component-helper';

.c-play-stop {}
.c-play-stop__button {}
