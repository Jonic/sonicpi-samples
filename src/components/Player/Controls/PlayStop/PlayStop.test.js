import React from 'react'
import { shallow } from 'enzyme'
import PlayStop from './PlayStop'

const isLoading = false
const isPlaying = false
const playStopClickHandler = jest.fn()

let wrapper

describe('<PlayStop />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <PlayStop
        isLoading={isLoading}
        isPlaying={isPlaying}
        playStopClickHandler={playStopClickHandler}
      />
    )
  })

  it('renders without crashing', () => {
    expect(wrapper.find('.c-PlayStop')).toBePresent()
  })
})
