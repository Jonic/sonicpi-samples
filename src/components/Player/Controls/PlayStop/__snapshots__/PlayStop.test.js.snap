// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayStop /> Renders... matches its snapshot 1`] = `
<div
  className="c-PlayStop"
>
  <button
    className="c-PlayStop__button"
    disabled={false}
    onClick={[Function]}
  >
    Play
  </button>
</div>
`;
